### PR TITLE
Fix tree table unit test

### LINKED ISSUE #15589: TreeTable Component: fix unit test
### Describe the bug

In this [Discussion](https://github.com/orgs/primefaces/discussions/1812) . I will also look into some broken tests and where needed extend test coverage.

This issue is not a bug ticket, but simply an issue for the PR to reference.

### Environment

windows

### Reproducer

_No response_

### Angular version

17.3.6

### PrimeNG version

17.16.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

v21.7.3

### Browser(s)

_No response_

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

### PR DESCRIPTION
**Pull request by dobanisola-scottlogic**
Fixes #15589

### CODE REVIEW

**Comment by nbrown-ScottLogic** [comment]
> @robGardiner01 @nbrown-ScottLogic @RogueTea Okay to review this?
Looks good thanks! Is the which depreciated? Why is it better to use code than which?

File: src/app/components/treetable/treetable.spec.ts [hunk 1]
```diff
@@ -631,7 +631,7 @@ class TestTreeTableComponent {
     ];
 }
 
-describe('TreeTable', () => {
+fdescribe('TreeTable', () => {
```
Comment (robGardiner01) [inline src/app/components/treetable/treetable.spec.ts:634]: Does this need to be changed back to just describe? Does fdescribe not make it only run this one?

File: src/app/components/slidemenu/slidemenu.spec.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { ButtonModule } from '../button/button';
 })
 class SlideMenuTestComponent {}
 
-describe('SlideMenu', () => {
+xdescribe('SlideMenu', () => {
```
Comment (robGardiner01) [inline src/app/components/slidemenu/slidemenu.spec.ts:16]: What does xdescribe do?

**Review by robGardiner01 — COMMENTED** [review]
Looks good! Just a couple questions about the describes

File: src/app/components/treetable/treetable.spec.ts (hunk 1 above)
Comment (dobanisola-scottlogic) [reply src/app/components/treetable/treetable.spec.ts:634]: Yes, I must have forgotten to change it.

**Comment by dobanisola-scottlogic** [comment]
> > @robGardiner01 @nbrown-ScottLogic @RogueTea Okay to review this?
> 
> Looks good thanks!
> 
> Is the which depreciated? Why is it better to use code than which?

@nbrown-ScottLogic Yes, which has been deprecated. https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/which
It's better to use key or code.

File: src/app/components/slidemenu/slidemenu.spec.ts (hunk 2 above)
Comment (RogueTea) [reply src/app/components/slidemenu/slidemenu.spec.ts:16]: The discussion is still open: https://github.com/orgs/primefaces/discussions/1938

**Comment by RogueTea** [comment]
Looks good to me.

**Pull request merged**